Source/KerbalCombatSystems/DataLink.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using KSP.UI.Screens;
+using UnityEngine;
+
```

Comment:
The module is applied to all parts that have the vanilla ksp communications module. The downside of this is that it's shared between relays and recievers. The basic moduledatalink will apply the other two based on part type in the same fashion as the weapon controllers.

The communications system itself will work in the same fashion as regular commnet for easy onboarding, both recievers and transmitters have a power level that varies by part and the square root of the distance will dictate if target lists can be successfully sent. The modules will just tell the shipcontroller about other scanning radii in the same way that baseline scanners just tell the ship how far it can see.

Reply:
Sounds good. Wouldn't it be safer and more easily configurable to specify the correct modules for each part in module manager though?

Reply:
This method would provide modded part support and it's a long list to individually outline, including the inbuilt antennas in command parts.

Reply:
Yeah good point